lib/streamlit/DeltaGenerator.py

```diff
@@ -1292,6 +1292,45 @@ def checkbox(self, element, ui_value, label, value=False):
         element.checkbox.value = current_value
         return current_value
 
+    @_widget
+    def multiselectbox(self, element, ui_value, label, options, format_func=str):
+        """Display a multiselect widget.
+        The multiselect widget starts as empty.
+
+        Parameters
+        ----------
+        label : str
+            A short label explaining to the user what this select widget is for.
+        options : list, tuple, numpy.ndarray, or pandas.Series
+            Labels for the select options. This will be cast to str internally
+            by default.
+        format_func : function
+            Function to modify the display of the labels. It receives the option
+            as an argument and its output will be cast to str.
+
+        Returns
+        -------
+        [str]
+            A list with the selected options
+
+        Example
+        -------
+        >>> options = st.multiselectbox(
+        ...     'What are your favorite colors',
+        ...     ('Green', 'Yellow', 'Red', 'Blue'))
+        >>>
+        >>> st.write('You selected:', options)
+
+        """
+
+        current_value = ui_value.value if ui_value is not None else []
+
+        element.multiselectbox.label = label
+        element.multiselectbox.value[:] = current_value  # should be a list
```

Comment:
I'll take care of this.

Reply:
Still I need to set this to `ui_value.value if ui_value`. I am renaming `element.multiselectbox.value` to `element.multiselectbox.default`, but not changing the logic. I am assuming if the state of the widget in the UI is set, this piece of code will read it (via the WidgetManager) and will send it back to the UI with that value.

Reply:
The default value should be set to the value the user passes in the function. We are not storing (only returning) the current state value, as it won't matter for the next run. 

P.S. Don't worry about it. Merge it this way. I will fix it later.

Reply:
@tvst, @kantuni and  have a doubt on the spec. I recall we said in Friday's meeting that we are ok for the multiselect not to have a default, since it may be awkward to initialize a multiselect with options, but I may have got it wrong.